api/python/provisioner/commands/validator/config_sections.py

```diff
@@ -0,0 +1,125 @@
+                output["cluster"] = data
+
+        output["node"] = servers
+        output["enclosure"] = enclosures
+
+        logger.debug(f"Parsed config output: {output}")
+
+        return output
+
+    def _validate_node_count(self, num_of_nodes, parsed):
+        """
+        Validates number of nodes.
+
+        Node count can only be 1 or multiples of 3.
+        There cannot be mismatch between count of
+        node sections and enclosure sections.
+
+        """
+        try:
+            logger.info(
+               f"Validating data for {num_of_nodes}-node(s) setup"
+            )
+
+            if not (int(num_of_nodes) == 1 or int(num_of_nodes) % 3 == 0):
+                raise ValueError(
+                  f"{num_of_nodes} nodes provided. "
+                  "Setup configuration can be done only on single node or "
+                  "in a cluster of node count in 3s"
+                )
+            else:
+                if ("srvnode_default" and not "enclosure_default" in parsed or
+                   "enclosure_default" and not "srvnode_default" in parsed):
+
+                    raise ValueError(
+                      "Default sections for servers and enclosures must be unianimous. "
+                      "Either both 'srvnode_default' and 'enclosure_default' sections "
+                      "must be present together or not."
+                    )
+
+                node_count = len(parsed["node"])
+
+                enclosure_count = len(parsed["enclosure"])
+
+                if node_count != int(num_of_nodes):
+                    raise ValueError(
+                      f"Node count mismatch! '{int(num_of_nodes)}' is given in input "
+                      f"whereas, data for '{node_count}'-node(s) is given in config file."
+                    )
+
+                elif not (node_count == 1 or node_count % 3 == 0):
+                    raise ValueError(
+                      f"Data for {node_count} nodes provided in config. "
+                      "Setup configuration can be done only on single node or "
+                      "in a cluster of node count in 3s"
+                    )
+
+                elif enclosure_count != node_count:
+                    logger.error(
+                        "ERROR: Invalid config data encountered. Count of node sections "
+                        "and enclosure sections in config file do not match."
+                    )
+                    raise ValueError(
+                       f"Number of enclosure sections: '{enclosure_count}' "
+                       f"does not match with number of node sections: '{node_count}'"
+                    )
+
+                else:
+                    logger.info(
+                        "Success: Node Validation done. "
+                        f"Applying config for {node_count}-node setup"
+                    )
+
+        except Exception as exc:
+            raise ValueError(f"Config Failed to apply: {str(exc)}")
```

Comment:
There are no new line here after the last code line.

Reply:
Done